src/credential/sd_jwt.rs

```diff
@@ -0,0 +1,364 @@
+use std::sync::Arc;
+
+use json_syntax::Object;
+use oid4vp::core::{
+    credential_format::ClaimFormatDesignation, presentation_definition::PresentationDefinition,
+};
+use ssi::{
+    claims::{
+        jwt::{AnyClaims, JWTClaims},
+        sd_jwt::{RevealedSdJwt, SdJwtBuf},
+        vc::v2::{Credential as _, JsonCredential},
+    },
+    prelude::AnyJsonCredential,
+};
+use uuid::Uuid;
+
+use crate::{CredentialType, KeyAlias};
+
+use super::{jwt_vc::JwtVc, Credential, ParsedCredential, ParsedCredentialInner};
+
+#[derive(Debug, uniffi::Object)]
+pub struct SdJwt {
+    pub(crate) id: Uuid,
+    pub(crate) key_alias: Option<KeyAlias>,
+    pub(crate) revealed_claims: JWTClaims,
+    pub(crate) credential: JsonCredential<Object>,
+    pub(crate) inner: SdJwtBuf,
+}
+
+// Internal utility methods for decoding a SdJwt.
+impl SdJwt {
+    /// Decode a SdJwt instance and return the revealed claims as a JSON value.
+    pub fn decode_reveal_json(&self) -> Result<serde_json::Value, SdJwtError> {
+        serde_json::to_value(&self.revealed_claims)
+            .map_err(|e| SdJwtError::Serialization(e.to_string()))
+    }
+
+    /// The types of the credential from the VCDM, excluding the base `VerifiableCredential` type.
+    pub fn types(&self) -> Vec<String> {
+        self.credential.additional_types().to_vec()
+    }
+
+    /// Returns the SD-JWT credential as an AnyCredential type.
+    pub fn credential(&self) -> AnyJsonCredential {
+        AnyJsonCredential::V2(self.credential.clone())
+    }
+
+    /// Check if the credential satisfies a presentation definition.
+    pub fn check_presentation_definition(&self, definition: &PresentationDefinition) -> bool {
+        // If the credential does not match the definition requested format,
+        // then return false.
+        if !definition
+            .format()
+            .contains_key(&ClaimFormatDesignation::Other("vcdm2_sd_jwt".into()))
+        {
+            return false;
+        }
+
+        let Ok(json) = serde_json::to_value(&self.credential) else {
+            // NOTE: if we cannot convert the credential to a JSON value, then we cannot
+            // check the presentation definition, so we return false.
+            //
+            // TODO: add logging to indicate that the credential could not be converted to JSON.
+            return false;
+        };
+
+        // Check the JSON-encoded credential against the definition.
+        definition.check_credential_validation(&json)
+    }
```

Comment:
Need to use the decoded and revealed claims, rather than the raw JSON credential which will contain `_sd` elements.

Reply:
See https://github.com/spruceid/mobile-sdk-rs/pull/43/files#diff-51db08be21682485c5a0ef28b2e22a4452e9b10c89659758fa4c4d000a0cf82eR231-R243

the `self.credential` value is referencing the decoded `JsonCredential`